Tests/Shared.Specs/StringAssertionSpecs.cs

```diff
@@ -1773,6 +1774,222 @@ public void When_the_actual_string_equivalent_to_the_expected_but_with_trailing_
+            Action act = () => "a".Should().ContainAll(new string[0] { });
+            Action actWithParams = () => "a".Should().ContainAll();
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            act
+                .ShouldThrow<ArgumentException>()
+                .WithMessage("Cannot assert string containment of values in empty collection");
+
+            actWithParams
+                .ShouldThrow<ArgumentException>()
+                .WithMessage("Cannot assert string containment of values in empty collection");
+        }
+
+        [Fact]
+        public void When_all_values_in_collection_are_present_in_string_containall_should_succeed()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            var because = Guid.NewGuid().ToString();
+            Action actWithBecause = () =>
+                "red green yellow".Should().ContainAll(new[] { "red", "green", "yellow" }, because: because);
+            Action actWithParams = () =>
+                "red green yellow".Should().ContainAll("red", "green", "yellow");
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            actWithBecause.ShouldNotThrow();
+            actWithParams.ShouldNotThrow();
+        }
+
+        [Fact]
+        public void When_none_of_the_values_in_collection_are_present_in_string_containall_should_throw_with_useful_info()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            var because = Guid.NewGuid().ToString();
+            Action actWithBecause = () =>
+                "red green yellow".Should().ContainAll(new[] { "blue", "orange", "pink" }, because: because);
+            Action actWithParams = () =>
+                "red green yellow".Should().ContainAll("blue", "orange", "pink");
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            const string expectedErrorMsg = "Expected string \"red green yellow\" to contain strings: " +
```

Comment:
Why don't you use interpolation strings?

Reply:
No problems. Happy to make the changes. Do you prefer the changes be added in new commits or rolled into the original commits (i.e. rewrite git history)? Also, what C# language level is acceptable? 7? (I played it cautious with string.Format etc. because I wasn't sure what your build process etc. was)

Reply:
I prefer an amend. And yes, C# 7 is perfectly fine.